.github/workflows/go-code-quality.yml

```
@@ -0,0 +1,34 @@
+name: Assess Go code quality
+
+on:
+  push:
+    paths:
+      - ".github/workflows/go-code-quality.yml"
+      - "cmd/**"
+      - "internal/**"
+
+jobs:
+  assess-go-code-quality:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
```

A lot of actions were updated to use latest javascript versions for its runtime.

```suggestion
      - uses: actions/checkout@v4
```

This update could be done on another PR, updating all actions to latest version, for ex.

---

I'll address that in a new PR. Thanks for the heads up 😉